rllib/algorithms/algorithm_config.py

```diff
@@ -873,13 +873,11 @@ def build_env_to_module_connector(self, env):
                 env, "single_observation_space", env.observation_space
             ),
             input_action_space=getattr(env, "single_action_space", env.action_space),
-            env=env,
```

Comment:
bug

Reply:
Correct, individual connector pieces or lists thereof are treated automatically.
I'm still not sure about the default piece at the end, though. I think we should have another config option to make this explicitly NOT included in the generated pipeline. ...

A test function would be tricky, I think. With the new design being so "stripped down" of all complexities, the user can simply setup a test script in which they:
* create the gym.Env
* call their config.build_env_to_module_connector(input_observation_space=..., ...)
* create their RLModule
* create a test episode object

Then they do `env.reset()`, store the results in the episode `.add_env_reset(...)`, call the connector `to_module = connector(rl_module=.., episodes=[episode], explore=True)`, then call the module's forward method: `module.forward_exploration(to_module)`

That's it.

I feel like if we do this in RLlib up front automatically, then we might already mess with some of the internals of a user-defined env, module, or connector, before the actual experiment begins. That's why I'm a little careful.

We should, however, produce meaningful error messages should the hand-overs between different connector pieces not work properly.

Most importantly, we need to have a dedicated page in the docs describing how simple it is to debug all this.

Reply:
I think for now the default is good as it is. Most modules use indeed the `SampleBatch.OBS` and eventually an additional `STATE_IN`. Even if `SampleBatch.NEXT_OBS` is used, this can be handled by `SampleBatch.OBS` (see `SACRLModule`). 

We might want to use here later on also something like the `get_input_specs`, `get_output_specs` and check for them via a decorator. What do you think? 

IAfter reading this, I agree with you on the pipeline testing. But we should give users then in the new documentation a guideline of how to do this on their own as you showed above. I am keeping book of all of these documentation TODOs in an `md` file, so we don't forget about them.